addons/mail/static/src/models/message_action_list.js

```diff
@@ -1,186 +1,195 @@
-                    this.hasReplyIcon,
-                    this.message.canBeDeleted,
-                    this.message.canBeDeleted,
-                    this.message.canStarBeToggled,
-                    this.message.hasReactionIcon,
-                ]
-                return actions.filter(Boolean).length;
+        _computeActionReplyTo() {
+            if (
+                this.messaging && this.messaging.inbox &&
+                this.message && !this.message.isTemporary && !this.message.isTransient &&
+                this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread && (
+                    this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread === this.messaging.inbox.thread ||
+                    this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread.channel
+                )
+            ) {
+                return insertAndReplace();
             }
             return clear();
         },
         /**
          * @private
-         * @returns {string}
+         * @returns {FieldCommand}
          */
-        _computeAddReactionText() {
-            return this.env._t("Add a Reaction");
+        _computeActionToggleCompact() {
+            if (this.messageView.isInChatWindow && (this.actionsWithoutCompactCount > this.compactThreshold)) {
+                return insertAndReplace();
+            }
+            return clear();
         },
         /**
          * @private
-         * @returns {boolean}
+         * @returns {FieldCommand}
          */
-        _computeHasCompactIcon() {
-            const COMPACT_THRESHOLD = 2;
-            const viewer = this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.threadViewer;
-            if (viewer && viewer.chatWindow) {
-                return this.actionsCount > COMPACT_THRESHOLD;
+        _computeActionToggleStar() {
+            if (this.message && this.message.canStarBeToggled) {
+                return insertAndReplace();
             }
+            return clear();
         },
         /**
          * @private
-         * @returns {boolean}
+         * @returns {FieldCommand}
          */
-        _computeHasMarkAsReadIcon() {
-            return Boolean(
-                this.messaging && this.messaging.inbox &&
-                this.messageView && this.messageView.messageListViewMessageViewItemOwner && this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread &&
-                this.messageView.messageListViewMessageViewItemOwner.messageListViewOwner.threadViewOwner.thread === this.messaging.inbox.thread
-            );
+        _computeFirstActionView() {
+            return replace(this.messageActionViews[0]);
```

Comment:
I think it's possible to technically have no message action views, so you must guard it:
```js
if (this.messageActionViews.length === 0) {
 return clear();
}
```